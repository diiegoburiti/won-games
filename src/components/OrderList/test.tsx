import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import itemsMock from './mock'

import OrderList from '.'

jest.mock('components/Empty', () => ({
  __esModule: true,
  default: function Mock() {
    return <div data-testid="Mock Empty" />
  }
}))

jest.mock('components/GameItem', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock GameItem">{children}</div>
  }
}))

describe('<OrderList />', () => {
  it('should render OrderList correclty', () => {
    const { container } = renderWithTheme(<OrderList items={itemsMock} />)

    expect(
      screen.getByRole('heading', { name: /My orders/i })
    ).toBeInTheDocument()

    expect(screen.getAllByTestId(/Mock GameItem/i)).toHaveLength(2)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render OrderList correclty', () => {
    renderWithTheme(<OrderList />)

    expect(screen.getByTestId(/Mock Empty/i)).toBeInTheDocument()
  })
})
