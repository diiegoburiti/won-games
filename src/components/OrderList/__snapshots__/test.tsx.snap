// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderList /> should render OrderList correclty 1`] = `
.c1 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c1::after {
  position: absolute;
  left: 0;
  bottom: -0.5rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c1::after {
  width: 3rem;
}

.c0 .sc-gsTCUz:last-child {
  border-bottom: 0;
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="black"
  >
    My orders
  </h2>
  <div
    data-testid="Mock GameItem"
  />
  <div
    data-testid="Mock GameItem"
  />
</div>
`
